pkg/login/social/generic_oauth.go

```diff
@@ -189,7 +191,12 @@ func (s *SocialGenericOAuth) UserInfo(client *http.Client, token *oauth2.Token)
 			return nil, fmt.Errorf("Error getting user info: %s", err)
 		}
 
-		err = json.Unmarshal(response.Body, &data)
+		if len(s.userInfoRootPath) != 0 {
+			user := gjson.Get(string(response.Body), s.userInfoRootPath).String()
```

Comment:
pasrsing, serializaing and then parsing again looks redundant, can't gjson do unmarshalling straight into &data?

Reply:
Gjson didn't provide such a method, and I also felt that it was not elegant enough.   😅